fvm/evm/handler/handler.go

```diff
@@ -134,45 +134,75 @@ func (h *ContractHandler) LastExecutedBlock() *types.Block {
 }
 
 // RunOrPanic runs an rlp-encoded evm transaction and
-// collects the gas fees and pay it to the coinbase address provided.
-func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, coinbase types.Address) {
+func (h *ContractHandler) RunOrPanic(rlpEncodedTx []byte, gasFeeCollector types.Address) {
 	// capture open tracing span
 	defer h.backend.StartChildSpan(trace.FVMEVMRun).End()
 
-	res, err := h.run(rlpEncodedTx, coinbase)
+	// capture coinbase init balance
+	cb := h.AccountByAddress(types.CoinbaseAddress, true)
+	initCoinbaseBalance := cb.Balance()
+
+	res, err := h.run(rlpEncodedTx)
 	panicOnErrorOrInvalidOrFailedState(res, err)
+
+	// transfer the gas fees collected to the gas fee collector address
+	afterBalance := cb.Balance()
+	diff := new(big.Int).Sub(afterBalance, initCoinbaseBalance)
+	if diff.Sign() > 0 {
+		cb.Transfer(gasFeeCollector, diff)
+	}
```

Comment:
what if `diff.Sign() < 0`, do we need to handle that case?

Reply:
It won't happen but I can add a warning if happen

Reply:
added an error for it.